src/lib/gitutils.ts

```diff
@@ -110,6 +110,21 @@ export const pushBranch = async (branchName: string): Promise<void> => {
   }
 };
 
+/**
+ * Tries to fetch the Git origin from an AzDo set environment variable, else falls back to fetching it from Git directly.
+ * @param absRepoPath The Absolute Path to the Repository to fetch the origin url.
+ */
+export const tryGetGitOrigin = async (
+  absRepoPath?: string
+): Promise<string> => {
+  return getAzdoOriginUrl().catch(_ => {
+    logger.warn(
+      "Could not get Git Origin for Azure DevOps - are you running 'spk' _not_ in a pipeline?"
+    );
+    return getOriginUrl(absRepoPath);
+  });
```

Comment:
I am not sure if you can write the code like this. :-) I thought it should be
```
  try {
    return await getAzdoOriginUrl();
  } catch (_) {
    logger.warn(
      "Could not get Git Origin for Azure DevOps - are you running 'spk' _not_ in a pipeline?"
    );
    return getOriginUrl(absRepoPath);
  });
```
or atleast, you need an await
```
  return await getAzdoOriginUrl().catch(_ => {
    logger.warn(
      "Could not get Git Origin for Azure DevOps - are you running 'spk' _not_ in a pipeline?"
    );
    return getOriginUrl(absRepoPath);
  });
```

maybe I am wrong.

Reply:
there's no need to return an `await` when you're returning a `Promise`  i think - @evanlouie can explain better

Reply:
ok thanks